src/main/java/org/terasology/gookeeper/system/HungerSystem.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.gookeeper.system;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.gookeeper.component.GooeyComponent;
+import org.terasology.gookeeper.component.HungerComponent;
+import org.terasology.gookeeper.event.GooeyFedEvent;
+import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.common.DisplayNameComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.physics.Physics;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+@Share(value = HungerSystem.class)
+public class HungerSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+
+    @In
+    private WorldProvider worldProvider;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private BlockEntityRegistry blockEntityRegistry;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    @In
+    private Physics physics;
+
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private PrefabManager prefabManager;
+
+    @In
+    private DelayManager delayManager;
+
+    @In
+    private NUIManager nuiManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(HungerSystem.class);
+    private Random random = new FastRandom();
+    private static final String eventID = "DECREASE_HEALTH_TICK";
+
+    @Override
+    public void initialise() {
```

Comment:
You may want to use the component added events for adding the first periodic action to a gooey.  

I *think* you may come into trouble with gooeys that get loaded in a future chunk after this system has been initialized.

Possibly a ```public void onGooeyActivated(OnAddedComponent event, EntityRef entity, GooeyComponent gooeyComponent, HungerComponent hungerComponent) {... }``` event handler?

Reply:
Good point 👍 I'll make the changes